ddl/lightning/message.go

```diff
@@ -0,0 +1,87 @@
+	LitErrRemoteDupExistErr string = "lightning: remote duplicate index key exist"
+	LitErrDiskQuotaLess     string = "lightning: specified disk quota is less than 100 GB disable the lightning"
+	LitErrIncompatiblePiTR  string = "lightning: the storage config log-backup.enable should be false when the lightning backfill is enabled"
+	LitErrExceedConcurrency string = "lightning: the concurrency is greater than lightning limit(tikv-importer.range-concurrency)"
+	LitErrUpdateDiskStats   string = "lightning: update disk usage error"
+	LitWarnEnvInitFail      string = "lightning: initialize environment failed"
+	LitWarnBackendNotExist  string = "lightning: backend not exist"
+	LitWarnConfigError      string = "lightning: build config for backend failed"
+	LitWarnGenMemLimit      string = "lightning: generate memory max limitation"
+	LitWarnExtentWorker     string = "lightning: extend worker failed will use worker count number worker to keep doing backfill task"
+	LitWarnDiskShortage     string = "lightning: local disk storage shortage"
+	LitInfoEnvInitSucc      string = "lightning: init global lightning backend environment finished"
+	LitInfoSortDir          string = "lightning: the lightning sorted dir"
+	LitInfoCreateBackend    string = "lightning: create one backend for an DDL job"
+	LitInfoCloseBackend     string = "lightning: close one backend for DDL job"
+	LitInfoOpenEngine       string = "lightning: open an engine for index reorg task"
+	LitInfoCleanUpEngine    string = "lightning: cleanUp one engine for index reorg task"
+	LitInfoCreateWrite      string = "lightning: create one local Writer for Index reorg task"
+	LitInfoCloseEngine      string = "lightning: flush all writer and get closed engine"
+	LitInfoDelEngine        string = "lightning: delete one engine"
+	LitInfoRemoteDupCheck   string = "lightning: start remote duplicate checking"
+	LitInfoStartImport      string = "lightning: start to import data"
+	LitInfoSetMemLimit      string = "lightning: set max memory limitation"
+	LitInfoChgMemSetting    string = "lightning: change memory setting for lightning"
+	LitInfoInitMemSetting   string = "lightning: initial memory setting for lightning"
+	LitInfoEngineDelete     string = "lightning: delete one engine from engine manager cache"
+	LitInfoUnsafeImport     string = "lightning: do a partial import data into the storage"
+	LitInfoDiskMaxLimit     string = "lightning: local disk storage usage arrive up limited"
+	LitInfoDiskQuotaChg     string = "lightning: local disk storage quota changed"
+)
+
+func logAllocMemFailedBackend(memRoot MemRoot, jobID int64) error {
+	logutil.BgLogger().Warn(LitErrAllocMemFail, zap.Int64("job ID", jobID),
+		zap.Int64("current memory usage", memRoot.CurrentUsage()),
+		zap.Int64("max memory quota", memRoot.MaxMemoryQuota()))
+	return errors.New(LitErrOutMaxMem)
+}
+
+func logAllocMemFailedEngine(memRoot MemRoot, jobID, idxID int64) error {
```

Comment:
`logXXX` looks like output a log, the returned error looks like there are something wrong during log. I assume whay you want to do is generate an error?